server/routes/auth.js

```diff
@@ -0,0 +1,68 @@
+const express = require('express')
+const router = express.Router()
+const bcrypt = require('bcrypt')
+const passport = require('passport')
+const LocalStrategy = require('passport-local').Strategy
+const User = require('../../usersDb')
+
+const salt = bcrypt.genSaltSync(10)
+const hash = (plainTextPassword) => bcrypt.hashSync(plainTextPassword, salt)
+
+passport.serializeUser((user, done) => {
+  done(null, user.id)
+})
+
+passport.deserializeUser((id, done) => {
+  User.findById(id)
+  .then((user) => {
+    return done(null, user)
+  })
+})
+
+passport.use(new LocalStrategy({
+    usernameField: 'email',
+    passwordField: 'password',
+    passReqToCallback: true
+  },
+  (request, username, password, done) => {
+    User.findUser(username)
+    .then((user) => {
+      if (!user[0]) {
+        return done(null, false, request.flash('loginError', 'Invalid username or password.'))
+      }
+      bcrypt.compare(password, user[0].password).then((result) => {
```

Comment:
This logic should not live inside this function. This function should be calling out to a function called `validatePassword` defined inside `db/user.js`, which will compare the salted password with the one stored in the database.